### ISSUE TITLE
MP4 Playback Problem with MP4Box

### ISSUE THREAD

**Issue by neptune84x** [staxrip/staxrip#1532]
Hello,
Some of the mp4 files I created with mp4box are working problematically. It does not rewind back and forth. It does not play fast. When I combine the same mp4 file with ffmpeg, it plays smoothly. I can't fully understand if it could be an optimization problem. 

I tested it on both my ios device and my webos lg tv. 

I am attaching the mediainfo file of two mp4 files I created with mp4box. I will try to understand what exactly is causing the problem. 

Robo0 file plays smoothly. Robo1 file is playing problematically. Both files are muxed with mp4box.

When the problematic file is muxed with ffmpeg, it also plays smoothly. 

I have to use mp4box because ffmpeg does not fully meet my needs. Thank you.

[Robo0.txt](https://github.com/user-attachments/files/17720921/Robo0.txt)
[Robo1.txt](https://github.com/user-attachments/files/17720924/Robo1.txt)

**Comment by neptune84x** [comment]
The iso6 code is additionally created in the problematic file, could it have something to do with this?

**Comment by Dendraspis** [comment]
The log files would be a lot more helpful than just the media info of the outcome....

**Comment by neptune84x** [comment]
I think the problem is in mp4box. Because when I mux the same file with ffmpeg, the problem is solved and the video is played without any problems.

**Comment by neptune84x** [comment]
The same situation occurred in another file. (1080p x265 non-HDR) When I mux with ffmpeg, the problem is solved. I don't know exactly how to identify the problem. It does not jump forward while playing the video and returns to the beginning again.

**Comment by Dendraspis** [comment]
Sure, but what shall I say without any information?

I would need the log file to see if StaxRip does something wrong and to reproduce it. If that's not the case, it's tool related and has nothing to do with StaxRip.

**Comment by neptune84x** [comment]
Where can I find the log file?

**Comment by neptune84x** [comment]
I muxed the same file separately with ffmpeg and mp4box. ffmpeg version works. mp4box version is problematic.

**Comment by neptune84x** [comment]
[New_ffmpeg_staxrip.log](https://github.com/user-attachments/files/17727988/New_ffmpeg_staxrip.log)
[New_mp4box_staxrip.log](https://github.com/user-attachments/files/17727989/New_mp4box_staxrip.log)

**Comment by neptune84x** [comment]
I hope these are the ones :)

**Comment by neptune84x** [comment]
If necessary, I can shoot a short video showing the problem and upload it to YouTube.

**Comment by Dendraspis** [comment]
Those are the log files of re-muxed files, that were encoded before. So the first question is: Why?

What I would need are 2 log files:  One from an Encode, that is muxed via MP4Box and one encode, that is mused via ffmpeg. Directly of course within one job (for both encodes) without re-muxing them.

Nevertheless it is also possible that the encoder is causing it. 
What do you think to get from `v3.0`? Are bugs, slower speed and poorer quality worth it?
In order to check it you should use the latest version that comes with StaxRip.

**Comment by neptune84x** [comment]
`Those are the log files of re-muxed files, that were encoded before. So the first question is: Why?`
My videos are mkv files. I mux them as mp4 to watch them on my tv and iOS device. I used to use Subler for this on macOS, so far the best Windows alternative is StaxRip.



`What I would need are 2 log files: One from an Encode, that is muxed via MP4Box and one encode, that is mused via ffmpeg. Directly of course within one job (for both encodes) without re-muxing them.`
I don't see any other log files, I think this is the one before the mux: 
[New_new_staxrip.log](https://github.com/user-attachments/files/17732493/New_new_staxrip.log)




```
Nevertheless it is also possible that the encoder is causing it.
What do you think to get from v3.0? Are bugs, slower speed and poorer quality worth it?
In order to check it you should use the latest version that comes with StaxRip.

```
I'm using the latest version of StaxRip. I'm sorry I don't understand what you mean by v3.0. It doesn't seem to be something I can fix myself. I apologize for that.

**Comment by Dendraspis** [comment]
Ahhhh, I see. So you keep the MKV files while you mux them to MP4 kust for compatibility reasons.

Do you have a short MP4 sample for me with that issue so I can see if I can reproduce the playback issue?

**Comment by neptune84x** [comment]
https://drive.google.com/drive/folders/1yMVs0efPlEMyW2LwWUNx2bCYxxO8tdF_?usp=sharing

the mkv one is the original file, the other two are the ones I muxed and I verified that the problem persists with this mp4box output. the ffmpeg version is fine. (but ffmpeg doesn't meet my needs)

**Comment by neptune84x** [comment]
Good news, I found a command that solved the problem. Everything is perfect now. I tested multiple files and my both devices.. :)

**Comment by Dendraspis** [comment]
I have tested several players and wanted to say, that I cannot reproduce it and you have to find a way to solve it, so I can adopt it if needed.  😅 

As `forcesync` is a pretty massive parameter, I can't implement it for all muxes:
<img width=600 src=https://github.com/user-attachments/assets/37981610-2819-4aaf-aae1-f4790c982de6>

So I will think about a solution ... stand by....

**Comment by neptune84x** [comment]
I will wait.. Thank you. Maybe you can add it to StaxRip as an optional option, such as the “-iPod” option.

**Comment by neptune84x** [comment]
It looks good :) I hope it will be used in the next version.

**Comment by Dendraspis** [comment]
It will be available in the next version `v2.42.3`,  but release will take a while.
Supporters will get it with the next RC version coming the next days.
